contrib/RL_out_sharp.lua

```diff
@@ -0,0 +1,208 @@
+      new_name = df.filename_increment(new_name)
+      -- limit to 50 more exports of the original export
+      if string.match(df.get_basename(new_name), "_%d%d$") == "_50" then break end
+      end
+
+    -- build the GMic command string
+    input_file = df.sanitize_filename(temp_name)
+    output_file = df.sanitize_filename(new_name)
+    options = " cut 0,255 round "
+    if df.get_filetype(temp_name) == "tif" then options = " -/ 256"..options end
+        
+    run_cmd = gmic.." "..input_file..gmic_operation..options.."o "..output_file..","..jpg_quality_str
+    
+    result = dtsys.external_command(run_cmd)
+    if result ~= 0 then
+      dt.print(_("sharpening error"))
+      return
+      end
+
+    -- delete temp image
+    os.remove(temp_name) 
+
+    end 
+  
+  dt.print(_("finished exporting"))
+  end
+
+-- new widgets ----------------------------------------------------------------
+
+output_folder_selector = dt.new_widget("file_chooser_button"){
+  title = _("select output folder"),
+  tooltip = _("select output folder"),
+  value = dt.preferences.read(MODULE_NAME, "output_folder", "string"),
+  is_directory = true,
+  changed_callback = function(self)
+    dt.preferences.write(MODULE_NAME, "output_folder", "string", self.value)
+    end
+  }
+
+sigma_slider = dt.new_widget("slider"){
+  label = _("sigma"),
+  tooltip = _("sigma parameter in RL algorithm"),
+  soft_min = 0.3,
+  soft_max = 1.0,
+  hard_min = 0.0,
+  hard_max = 3.0,
+  step = 0.05,
+  digits = 2,
+  value = 0.7
+  }
+
+iterations_slider = dt.new_widget("slider"){
+  label = _("iterations"),
+  tooltip = _("increase for better sharpening, but slower"),
+  soft_min = 0,
+  soft_max = 100,
+  hard_min = 0,
+  hard_max = 100,
+  step = 5,
+  digits = 0,
+  value = 10.0
+  }
+
+jpg_quality_slider = dt.new_widget("slider"){
+  label = _("output jpg quality"),
+  tooltip = _("quality of the output jpg file"),
+  soft_min = 70,
+  soft_max = 100,
+  hard_min = 70,
+  hard_max = 100,
+  step = 2,
+  digits = 0,
+  value = 95.0
+  }
+
+local storage_widget = dt.new_widget("box"){
+  orientation = "vertical",
+  output_folder_selector,
+  sigma_slider,
+  iterations_slider,
+  jpg_quality_slider
+  }
+
+-- register new storage -------------------------------------------------------
+dt.register_storage("exp2RL", _("RL output sharpen"), nil, export2RL, supported, save_preferences, storage_widget)
+
+-- register the new preferences -----------------------------------------------
+dt.preferences.register(MODULE_NAME, "gmic_exe", "file", 
+_("executable for GMic CLI"), 
+_("select executable for GMic command line version")  , _("(None)"))
+
+-- set sliders to the last used value -----------------------------------------
+sigma_slider.value = dt.preferences.read(MODULE_NAME, "sigma", "float")
+iterations_slider.value = dt.preferences.read(MODULE_NAME, "iterations", "float")
+jpg_quality_slider.value = dt.preferences.read(MODULE_NAME, "jpg_quality", "float")
+
```

Comment:
These statements will run when you first load the script, but not after so they will never update.  They need to be moved into the export2RL function if you want to save the values that were used.